RecoPixelVertexing/PixelLowPtUtilities/plugins/PixelClusterShapeExtractor.cc

```diff
@@ -0,0 +1,407 @@
+// VI January 2012: needs to be migrated to use cluster directly
+
+#include "FWCore/Framework/interface/global/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+#include "SimTracker/TrackerHitAssociation/interface/TrackerHitAssociator.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelClusterShapeCache.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "RecoPixelVertexing/PixelLowPtUtilities/interface/ClusterShapeHitFilter.h"
+
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#
+
+#include "TROOT.h"
+#include "TFile.h"
+#include "TH1F.h"
+#include "TH2F.h"
+
+#include <map>
+#include <vector>
+#include <fstream>
+
+#include<mutex>
+
+using namespace std;
+
+// pixel
+#define exMax 10
+#define eyMax 15
```

Comment:
CMS coding rules say not to use macros for constants
.. will not be enforced in /test